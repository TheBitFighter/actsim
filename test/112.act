deftype struct(int<4> x, y) { }


defproc sink(chan?(struct) X)
{
   struct s;
   chp {
     *[ X?s; log ("got ", s.x, " and ", s.y) ]
   }
}

defproc src (chan!(struct) O)
{
  struct t;
  chp {
    t.x := 1, t.y := 2;
    O!t;
    t.x := 2, t.y := 3;
    O!t;
    skip
  }
}
  

defproc test()
{
  src x;
  sink y(x.O);
}
