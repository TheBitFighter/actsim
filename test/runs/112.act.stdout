[                  20] <y>  got 1 and 2
[                  40] <y>  got 2 and 3
